android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/MobileWalletAdapter.kt

```diff
@@ -0,0 +1,187 @@
+package com.solana.mobilewalletadapter.clientlib
+
+import android.net.Uri
+import android.util.Log
+import androidx.activity.result.ActivityResultCaller
+import com.solana.mobilewalletadapter.clientlib.protocol.JsonRpc20Client
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient
+import com.solana.mobilewalletadapter.clientlib.scenario.LocalAssociationScenario
+import com.solana.mobilewalletadapter.clientlib.scenario.Scenario
+import com.solana.mobilewalletadapter.clientlib.scenario.ScenarioChooser
+import com.solana.mobilewalletadapter.clientlib.scenario.ScenarioTypes
+import com.solana.mobilewalletadapter.common.ProtocolContract
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.IOException
+import java.util.concurrent.CancellationException
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+fun LocalAssociationScenario.begin(): MobileWalletAdapterClient {
+    return start().get(MobileWalletAdapter.ASSOCIATION_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+}
+
+fun LocalAssociationScenario.end() {
+    close().get(MobileWalletAdapter.ASSOCIATION_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+}
+
+@Suppress("BlockingMethodInNonBlockingContext")
```

Comment:
My preference is always to suppress at the smallest scope possible. Can this be applied to the statement or function instead?

Reply:
Yes it can. Will do.